lib/segment/src/common/rocksdb_buffered_update_wrapper.rs

```diff
@@ -16,16 +16,20 @@ use crate::common::Flusher;
 #[derive(Debug)]
 pub struct DatabaseColumnScheduledUpdateWrapper {
     db: DatabaseColumnWrapper,
-    deleted_pending_persistence: Mutex<HashSet<Vec<u8>>>,
-    insert_pending_persistence: Mutex<HashMap<Vec<u8>, Vec<u8>>>,
+    pending_operations: Mutex<PendingOperations>, // in-flight operations persisted on flush
+}
+
+#[derive(Debug, Default)]
+struct PendingOperations {
+    deleted: HashSet<Vec<u8>>,
+    inserted: HashMap<Vec<u8>, Vec<u8>>,
```

Comment:
do we even need parallel structures here? Wouldn't it be easier to represent as hashmap of enum?

Reply:
Do you mean something like this? :thinking: 

```rs
enum Operation {
    Inserted(Vec<u8>),
    Deleted,
}

type PendingOperations = HashMap<Vec<u8>, Operation>;
```

Reply:
Yes.

Insertions have priority over deletions, so a delete entry can just be replaced. If we really want we can add the same debug assertion in a different place.

Reply:
I'd prefer just tackling the Mutex refactoring in the current PR. 
Using a single data structure is a different transformation that should be done in isolation.